### DISCUSSION TITLE
Clarification on "Passes" on round_data_files

### DISCUSSION THREAD

**Discussion by boshington** [jatkinson1000/archeryutils#95] [Q&A] [answered]
Howdy, 

I'm trying to understand what a "pass" is in terms of the round data files. For example, why does an agb_indoor portsmouth consist of 2 passes of 30 arrows instead of one pass of 60 arrows when all other attributes are the same. 

I'm creating a scorecard companion app (mostly for my own use) and want to make sure I'm not missing anything significant.

All the best 
Tom

Edit: Have noted that there's a description in rounds.py that defines a pass as a sub-unit of a round, but still not sure if there is a benefit for using passes for same-distance rounds.

**Comment by jatkinson1000** [comment, marked as answer]
Hi @boshington Really glad the code is of use to someone 😅 

So - the pass is a subset of a round.

If you think of the 1440 you have 4 passes of 3-dozen - one at each distance 90m, 70m, 50m, 30m on the 90m round.
If you think of the WA720 you still have 2 passes of 3-dozen, but both are at 70m.

With the indoor rounds a similar structure is followed, with each half of a round being a 2.5-dozen pass.
This is consistent with how Ianseo does things (for the WA18 at least - you can get a different scorecard for each half).

Hope that makes sense, but if not please ask further!

**Reply by boshington** [reply]
Thanks, just a way to break down rounds into smaller chunks. 

I've only been shooting since april, so I'm not super familiar with competition level shooting.
Also, I wasn't aware of ianseo, it looks interesting, but more complex than I would have expected. 

Thanks for your help.

All the best
Tom

**Reply by jatkinson1000** [reply]
No problem - feel free to come back with any other questions, suggestions, or contributions :)

Ianseo is very widely used, but the software is... "a pit of monsters".

I'd love to see what you produce at some point.
One of the better-known uses of this code is https://archerycalculator.co.uk/ :)
